components/extension-mgt/org.wso2.carbon.identity.extension.mgt/src/test/java/org/wso2/carbon/identity/extension/mgt/TestUtils.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.extension.mgt;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.wso2.carbon.identity.extension.mgt.utils.ExtensionMgtConstants;
+
+/**
+ * Utility functions for testing components of the extension management service.
+ */
+public class TestUtils {
```

Comment:
Why do we need to have a separate test util class? We can have these as private methods inside the needed test class as we do not have any other classes in this module that can use this.

Reply:
There are two other classes that should also be unit tested: `ExtensionStore` and `ExtensionManager`. Since the template reading functionality is implemented in the OSGI activate method and is private, we need to read the JSON files separately and provide them to these classes. Thus, having this Util class to reuse the functionality is beneficial. Additionally, there are two other utility methods that should be added to this Util class to read `template.json` and `metadata.json`. Therefore, this approach is scalable IMO.